opal/corelib/string/unpack.rb

```diff
@@ -686,6 +686,33 @@ class String
       'p': null
     }
 
+    function toArray(callback) {
+      return function(data) {
+        return [callback(data)];
+      }
+    }
+
+    var optimized = {
+      'C*': identityFunction,
+      'c*': toNByteSigned(1, identityFunction),
+      'A*': wrapIntoArray(joinChars(bytesToAsciiChars(filterTrailingZerosAndSpaces(identityFunction)))),
+      'a*': wrapIntoArray(joinChars(bytesToAsciiChars(identityFunction))),
+      'M*': toArray(qpdecode(joinChars(bytesToAsciiChars(identityFunction)))),
+      'm*': toArray(base64Decode(joinChars(bytesToAsciiChars(identityFunction)))),
```

Comment:
I've added a `toArray` function because the specs on `M*` and `m*` expect an Array.

Reply:
Oh! I've missed the `wrapIntoArray` function... composition FTW! 🤣🤣🤣